spacy_loggers/clearml.py

```diff
@@ -25,113 +119,150 @@ def clearml_logger_v1(
-                                # This saves a lot of space in the UI
-                                task.get_logger().report_scalar(
-                                    title=metric,
-                                    series=sub_metric,
-                                    iteration=info["step"],
-                                    value=sub_metric_value,
-                                )
-                        elif isinstance(metric_value, (float, int)):
-                            task.get_logger().report_scalar(
-                                metric,
-                                metric,
-                                iteration=info["step"],
-                                value=metric_value,
-                            )
-                if model_log_interval and info.get("output_path"):
-                    if info["step"] % model_log_interval == 0 and info["step"] != 0:
-                        if log_latest_dir:
-                            last_model.update_weights_package(
-                                weights_path=log_latest_dir,
-                                auto_delete_file=False,
-                                target_filename="last_model",
-                            )
-                        if (
-                            log_best_dir
-                            and info["score"] == max(info["checkpoints"])[0]
-                        ):
-                            best_model.update_weights_package(
-                                weights_path=log_best_dir,
-                                auto_delete_file=False,
-                                target_filename="best_model",
-                            )
-
-        def finalize() -> None:
-            console_finalize()
-            task.flush(wait_for_uploads=True)
-            task.close()
+            elif isinstance(metric_value, (float, int)):
+                task.get_logger().report_scalar(
+                    metric,
+                    metric,
+                    iteration=info["step"],
+                    value=metric_value,
+                )
 
-        return log_step, finalize
+    if model_log_interval and info.get("output_path"):
+        if info["step"] % model_log_interval == 0 and info["step"] != 0:
+            if log_latest_dir:
+                assert last_model is not None
+                last_model.update_weights_package(
+                    weights_path=log_latest_dir,
+                    auto_delete_file=False,
+                    target_filename="last_model",
+                )
+            if log_best_dir and info["score"] == max(info["checkpoints"])[0]:
+                assert best_model is not None
+                best_model.update_weights_package(
+                    weights_path=log_best_dir,
+                    auto_delete_file=False,
+                    target_filename="best_model",
+                )
 
-    return setup_logger
+
+def _finalize_clearml(
+    task: Any,
+):
```

Comment:
Is this how `black` formats the signature?

Reply:
Interestingly, `black` kept the above formatting until I changed it be an one-liner.

Reply:
Yeah, it's a bit weird sometimes.